### DISCUSSION TITLE
[FeatureRequest]: Cancel chained requests when a parent doesn't return values

### DISCUSSION THREAD

**Discussion by mertbakir** [grafana/grafana#38331] [Data Sources]
**How to reproduce it (as minimally and precisely as possible)**:

1. Create a variable that returns string, let's call this: ```$name```

```
SELECT "name" 
FROM table
WHERE 
some condition
``` 

2. Make it multi-selection and Include All Option.

3. Create another variable, and call this: ```$value```

```
SELECT "value"
FROM table
WHERE
"name" in ($name)
```

When $name returns nothing, (the conditions are not satisfied for it or when we select "None" from the dropdown.):
- The query for the dependent variable ```$value``` will return an error. 
- Normally, with "multi-selection" Grafana adds apostrophes and separates each record with a comma.
- When there is no value selected/returned for the "multi-selection" variable, Grafana still should set the variable an empty string ```''```

The Error When Nothing Returned From the variable $name
> pq: syntax error at or near ")"

Edit: This happens only when the variable returns nothing and "All" selected. Setting a ```custom all value``` = ```''``` can fix the situation, but then "All" option won't work as intended when there are "multiple options" and "all" selected.

I can't post screenshots but I hope this was clear enough. You should look into when the variable returns nothing and "All" selected.

I've tried the advanced options explained in the docs: https://grafana.com/docs/grafana/latest/variables/advanced-variable-format-options/ No luck.

**Environment**:
- Grafana version: 7.3.5
- Data source type & version: Postgres 9
- OS Grafana is installed on: Windows 10 64 bit
- User OS & Browser: Windows 10 64 bit & Chrome
- Grafana plugins: -
- Others: -

**Comment by ying-jeanne** [comment]
the bug reported is reproduced when 2 variables. variable 2 is depending on the result of variable 1, as variable 1 is unselected, we got error 400 bad requests. we may want to skip query for variable 2 when variable 1 has no result and set variable 2 to unselected directly 

![image](https://user-images.githubusercontent.com/74549700/103889087-1d58ef80-50e6-11eb-9087-3c7857335ac7.png)

**Comment by hugohaggmark** [comment]
I'm going to change this to a feature request because the current variable system is not designed to cancel dependent queries when the parent variable doesn't return values. Not sure this is something that can be easily accomplished.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gabor** [comment]
(i am removing the postgres-label from this discussion because it seems the problem is related to the dashboard-variables, not to postgres. if that's not the case, just ping me and we can add it back. thanks!)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
